Lesson39/savas/src/components/__test__/BackgroundColorChanger.test.js

```diff
@@ -0,0 +1,20 @@
+import React from "react";
+import { render, fireEvent } from "@testing-library/react";
+import BackgroundColorChanger from "../BackgroundColorChanger";
+
+test("Background color changes on button click", () => {
+  const { getByText, container } = render(<BackgroundColorChanger />);
+
+  const greenButton = getByText("Green");
```

Comment:
Avoid getting elements by text when testing, this is an unreliable approach. Instead, add data-testid attributes as we did in the lesson and use `screen.getByTestId('your-data-testid)`.

Reply:
Here, if I use screen.getByTestId, the test gives 1 fail. I couldn't understand why it is?